File: src/pewmode/Spritable.java
package pewmode;

 interface Spritable {
     float getxPos();

     void setxPos(float xPos);

     float getyPos();

     void setyPos(float yPos);

     float getxSpeed();

     void setxSpeed(float xSpeed);

     float getySpeed();

     void setySpeed(float ySpeed);

     void setSpeed(float speed);

     float getSpeed();

     void update();

     void render();

     void init();

     void move(float x, float y);

     void blindMove(float x, float y);

     void setAlpha(float i);

     void setScale(float v);

 }

File: src/pewmode/Spritable.kt
package pewmode

internal interface Spritable {
    fun getxPos(): Float

    fun setxPos(xPos: Float)

    fun getyPos(): Float

    fun setyPos(yPos: Float)

    fun getxSpeed(): Float

    fun setxSpeed(xSpeed: Float)

    fun getySpeed(): Float

    fun setySpeed(ySpeed: Float)

    var speed: Float

    fun update()

    fun render()

    fun init()

    fun move(x: Float, y: Float)

    fun blindMove(x: Float, y: Float)

    fun setAlpha(i: Float)

    fun setScale(v: Float)

    fun setMenuMode(menu: Boolean)
}
